core/src/main/scala/org/apache/spark/sql/execution/joins/LocalJoin.scala

```diff
@@ -335,10 +410,15 @@ case class LocalJoin(leftKeys: Seq[Expression],
           streamSideKeys.map(BindReferences.bindReference(_, left.output)))
     }
     val streamKeyVars = ctx.generateExpressions(streamKeys)
-    mapAccessor.generateMapLookup(entryVar, localValueVar, keyIsUniqueTerm,
+
+    val mapAccesCode = mapAccessor.generateMapLookup(entryVar, localValueVar, keyIsUniqueTerm,
       numRowsTerm, nullMaskVars, initCode, checkCondition,
       streamSideKeys, streamKeyVars, buildKeyVars, buildVars, input,
       resultVars, dictionaryArrayTerm, joinType)
+    s"""
+       $evaluatedInputVars
+       $mapAccesCode
+     """
```

Comment:
Ok, the issue reported by Sonal is similar. Both this and that one are due to the same change. For dictionary optimization I had changed the "usedInputs" method to return empty set with the expectation that code will instantiate required variables instead of expanding everything upfront in the consume() call. I modified it back for both the classes and checked the perf of TPCETest Q1 is down by 15-20% (160ms vs 190ms best, 180ms vs 210ms average). While not huge, this is still significant.

The thing I tried (which worked in SnappyHashAggregateExec) is to explicitly invoke: evaluateRequiredVariables(consumer.output, input, consumer.references) before consume is invoked which calls the child consume methods. Likewise you can try invoking this (you will have to pass input) in each of the "gen*JoinCodes" method in ObjectHashMapAccessor after the continue due to join failure but before the consumeResult is emitted. So, for example, for inner join you can place this before or after the "declareLocalVars" in emitted code, same in outer join though it won't matter there since there is no case of continue, or before consumeResult in anti-join. Did a quick check with inner-join that Q1 perf is back to original level, so if this works for the queries you saw failures for then it will be the more performant option.